flink_jobs_v2/batch_status/src/test/java/argo/batch/ArgoStatusBatchTest.java

```diff
@@ -139,8 +138,20 @@ public void testMain() throws Exception {
         Assert.assertTrue(explpdata.containsAll(resultlpdata));
 
         
+        //************* Test unioned metric data of previous and current date ************
+          DataSet<MetricData> mdataPrevTotalDS = mdataDS.union(pdataMin);
+        URL unionpdataURL = ArgoStatusBatchTest.class.getResource("/test/uniondata.avro");
+
+        Path unionpin = new Path(unionpdataURL.getPath());
+        AvroInputFormat<MetricData> unionpdataAvro = new AvroInputFormat<MetricData>(unionpin, MetricData.class);
+
+        DataSet<MetricData> unionpdataDS = env.createInput(unionpdataAvro);
+
+        Assert.assertTrue(unionpdataDS.collect().containsAll(mdataPrevTotalDS.collect()));
+      
+        
         
-        pdataMin.writeAsText("../resources/test/output/results.txt", FileSystem.WriteMode.OVERWRITE).setParallelism(1);
+        mdataPrevTotalDS.writeAsText("../resources/test/output/results.txt", FileSystem.WriteMode.OVERWRITE).setParallelism(1);
```

Comment:
Do we need this stored in a file? Can we check the data in memory ?

Reply:
We need a datasink else the test is not executed. Replaced the Dataset.writeAsText() with adding a sink by Dataset.ouput() with a DiscardingOutputFormat